### PR TITLE
* register GitHub repo in `initialise()`

### PR DESCRIPTION
**Pull request by soniamitchell**
* save latest commit hash and remote repo in working config
* ensure repo is clean

### CODE REVIEW

File: content/docs/interface/fdp/_index.md [hunk 1]
```diff
@@ -29,13 +29,15 @@ fair push config.yaml
...
   - If no version is given, then one should be written such that patch is incremented if the data product already exists, otherwise version should be set to 0.0.1.
   - `register:` is removed and external objects / data products are written in `read:`
-- save the working *config.yaml* file in the local data store (in *<local_store>/coderun/\<date>-\<time>/config.yaml*) 
-- save the submission script to the local data store in *<local_store>/coderun/\<date>-\<time>/script.sh* (note that *config.yaml* should contain either `script:` that should be saved as the submission script, or `script_path:` that points to the file that should be saved as the submission script) 
-- save the path to *<local_store>/coderun/\<date>-\<time>/* in the global environment as `$FDP_CONFIG_DIR` so that it can be picked up by the script that is run after this has been completed
 - `local_repo:` must always be given in the *config.yaml* file
+  - ensure the repo is clean
+  - get the hash of the latest commit
   - get the remote repo url from the local repo
```
Comment (richardreeve) [inline content/docs/interface/fdp/_index.md:35]: if it's not already provided in the user's config.yaml?

File: content/docs/interface/fdp/_index.md [hunk 2]
```diff
@@ -29,13 +29,15 @@ fair push config.yaml
...
-  - get the hash of the latest commit on GitHub and store this in the data registry, associated with the submission script `storage_location` (this is where the script should be stored)
+  - add these to the working *config.yaml* file
   - note that there are exceptions and the user may reference a script located outside of a repository
+- save the working *config.yaml* file in the local data store (in *<local_store>/coderun/\<date>-\<time>/config.yaml*)
+- save the submission script to the local data store in *<local_store>/coderun/\<date>-\<time>/script.sh* (note that *config.yaml* should contain either `script:` that should be saved as the submission script, or `script_path:` that points to the file that should be saved as the submission script)
+- save the path to *<local_store>/coderun/\<date>-\<time>/* in the global environment as `$FDP_CONFIG_DIR` so that it can be picked up by the script that is run after this has been completed
```
Comment (richardreeve) [inline content/docs/interface/fdp/_index.md:40]: If we're adding date and time, we should probably stick the [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) and use "*\<date>T\<time>*".

**Review by richardreeve — COMMENTED** [review]
A couple of minor points

File: content/docs/interface/fdp/_index.md (hunk 1 above)
Comment (soniamitchell) [reply content/docs/interface/fdp/_index.md]: done

File: content/docs/interface/fdp/_index.md (hunk 2 above)
Comment (soniamitchell) [reply content/docs/interface/fdp/_index.md]: done

**Review by richardreeve — APPROVED** [review]

**Pull request merged**